source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -921,13 +913,39 @@ namespace
         doc->removeChild(uniqueName);
     }
 
-    void writeNodeGraph(const PvtPrim * src, DocumentPtr dest)
+    void writeNodeGraph(const PvtPrim* src, DocumentPtr dest)
     {
         NodeGraphPtr destNodeGraph = dest->addNodeGraph(src->getName());
         writeMetadata(src, destNodeGraph, nodegraphMetadata);
 
         RtNodeGraph nodegraph(src->hnd());
 
+        // Write inputs/parameters.
```

Comment:
Turns out inputs and parameters are already supported explicitly for NodeGraph in core API. So just need to write the data out at export time.

Reply:
Import already supported reading this in.

Reply:
For the comment about support, does this mean MTLX nodegraph or Runtime nodegraph. Pretty sure this will fail validation for MTLX load.

Reply:
I mean MTLX nodegraph. It works already since NodeGraph derives from InterfaceElement. And it passes validation. The test "Runtime: FileIo NodeGraph" will both read and write a document with this content, and `RtFileIo::read` does document validation..